Entities/Structures/Door/DoorCommon.as

```diff
@@ -7,16 +7,35 @@ bool canOpenDoor(CBlob@ this, CBlob@ blob)
 	        (this.getTeamNum() == 255 || this.getTeamNum() == blob.getTeamNum()) &&
 	        (blob.hasTag("player") || blob.hasTag("vehicle") || blob.hasTag("migrant"))) //tags that can open doors
 	{
-		Vec2f direction = Vec2f(0, -1);
-		direction.RotateBy(this.getAngleDegrees());
-
 		Vec2f doorpos = this.getPosition();
-		Vec2f playerpos = blob.getPosition();
-
-		if (blob.isKeyPressed(key_left) && playerpos.x > doorpos.x && Maths::Abs(playerpos.y - doorpos.y) < 11) return true;
-		if (blob.isKeyPressed(key_right) && playerpos.x < doorpos.x && Maths::Abs(playerpos.y - doorpos.y) < 11) return true;
-		if (blob.isKeyPressed(key_up) && playerpos.y > doorpos.y && Maths::Abs(playerpos.x - doorpos.x) < 11) return true;
-		if (blob.isKeyPressed(key_down) && playerpos.y < doorpos.y && Maths::Abs(playerpos.x - doorpos.x) < 11) return true;
+		Vec2f blobpos = blob.getPosition();
+	
+		if (blob.hasTag("vehicle"))
+		{
+			if (doorpos.y > blobpos.y + blob.getHeight()/2 + 2) return false;
+			
+			AttachmentPoint@[] aps;
+			if (blob.getAttachmentPoints(@aps))
+			{
+				for (u8 i = 0; i < aps.length; i++)
+				{
+					AttachmentPoint@ ap = aps[i];
+					if (ap.name != "ROWER" && ap.name != "DRIVER") continue;
```

Comment:
```suggestion
					if (ap.name != "ROWER" && ap.name != "DRIVER" && ap.name != "SAIL") continue;
```

Otherwise driving alone in the longboat sail seat does not allow you to go through doors.